proxy/src/main/java/com/wavefront/agent/channel/ChannelUtils.java

```diff
@@ -97,6 +82,53 @@ public static void writeHttpResponse(final ChannelHandlerContext ctx,
                                        final HttpResponseStatus status,
                                        final Object /* JsonNode | CharSequence */ contents,
                                        boolean keepAlive) {
+    writeHttpResponse(ctx, makeResponse(status, contents), keepAlive);
+  }
+
+  /**
+   * Writes HTTP response back to client.
+   *
+   * @param ctx       Channel handler context.
+   * @param response  HTTP response object.
+   * @param request   HTTP request object (to extract keep-alive flag).
+   */
+  public static void writeHttpResponse(final ChannelHandlerContext ctx,
+                                       final HttpResponse response,
+                                       final HttpMessage request) {
+    writeHttpResponse(ctx, response, HttpUtil.isKeepAlive(request));
+  }
+
+  /**
+   * Writes HTTP response back to client.
+   *
+   * @param ctx       Channel handler context.
+   * @param response  HTTP response object.
+   * @param keepAlive Keep-alive requested.
+   */
+  public static void writeHttpResponse(final ChannelHandlerContext ctx,
+                                       final HttpResponse response,
+                                       boolean keepAlive) {
+    // Decide whether to close the connection or not.
+    if (keepAlive) {
+      // Add keep alive header as per:
+      // - http://www.w3.org/Protocols/HTTP/1.1/draft-ietf-http-v11-spec-01.html#Connection
+      response.headers().set(HttpHeaderNames.CONNECTION, HttpHeaderValues.KEEP_ALIVE);
+      ctx.write(response);
+    } else {
+      ctx.write(response);
+      ctx.writeAndFlush(Unpooled.EMPTY_BUFFER).addListener(ChannelFutureListener.CLOSE);
```

Comment:
can this just be collapsed into a single writeAndFlush() call?

Reply:
yup.